File: App/mindspace_api/app/Models/TherapistProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class TherapistProfile extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'user_id',
        'profile_picture_path',
        'education_history',
        'hourly_rate',
        'experience_years',
        'specializations',
        'problem_areas',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'specializations' => 'array',
    ];

    /**
     * Get the user that owns the therapist profile.
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: App/mindspace_app/lib/models/therapist_profile.dart
class TherapistProfile {
  final String profilePicturePath;
  final String educationHistory;
  final int hourlyRate;
  final int experienceYears;
  final List<String> specializations;
  final String problemAreas;

  TherapistProfile({
    required this.profilePicturePath,
    required this.educationHistory,
    required this.hourlyRate,
    required this.experienceYears,
    required this.specializations,
    required this.problemAreas,
  });

  factory TherapistProfile.fromJson(Map<String, dynamic> json) {
    return TherapistProfile(
      profilePicturePath: json['profile_picture_path'],
      educationHistory: json['education_history'],
      hourlyRate: json['hourly_rate'],
      experienceYears: json['experience_years'],
      specializations: List<String>.from(json['specializations']),
      problemAreas: json['problem_areas'],
    );
  }
}